### PR TITLE
Fixing a number of open issues. All APIs now ready for testing

### PR DESCRIPTION
**Pull request by mosopeogundipe**
Issues Fixed:
1) Issue #18: rewrote PUT /checkoutMovies API to call search API if movie isn't present in Faves team inventory. If the search API returns a result for the movie id, we add the movie in the Faves team inventory and allow the checkout to go through.
2) general cleaning of GET/checkout and PUT/checkout APIs
3) Issue #23: Enable CORS for our APIs on controller level
4) Issue #20: Updated api path from PUT /checkOutMovies to PUT /checkOutMovie to
accurately reflect the functionality
5) Issue #20: Updated API paths to /checkOutMovie and /returnMovie in IncludeFilters.java

Test Env: https://hsit-fav-che-issue-reso-zdspwy.herokuapp.com/
**Example Request** 
Request Method: `PUT`
URI: https://hsit-fav-che-issue-reso-zdspwy.herokuapp.com/checkOutMovie
RequestBody:
```
{
	"userId":2,
	"movieId":80211
}
```

P.S: I could have made one more modification but it's not that important, IMO. So, I skipped it.

### CODE REVIEW

File: src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java [hunk 1]
```diff
@@ -242,54 +242,47 @@ public OperationalResponse favoriteMovie(OperationalRequest request) {
...
+			s.add(movieId);
+			SearchMoviesResponse searchAPIResp = APIClient.getAllMovies(s);
+			if(searchAPIResp == null) {
+				resp.setMessage("Search API has no record for Movie with Id " + movieId);
+				return ResponseEntity.status(HttpStatus.NOT_FOUND).body(resp);
+			}
+			record = new Inventory(movieId, DEFAULT_NUM_OF_MOVIES, DEFAULT_NUM_OF_MOVIES);
 		}
-		Inventory record = inventory.get();
+		record = (record == null) ? inventoryRepository.getOne(movieId) : record;
```
Comment (kedarkhetia) [inline src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java]: Can we do something like below? This way we will save 1 extra DB call. Also, `getOne(movieId)` throws unchecked exception `EntityNotFoundException` if the entity is not found. We should try and avoid any underlying code which throws unchecked exceptions. 

```
record = inventoryRepository.findById(movieId);
if(recode == null) {
    HashSet<Integer> s = new HashSet<Integer>();
    s.add(movieId);
    SearchMoviesResponse searchAPIResp = APIClient.getAllMovies(s);
    if(searchAPIResp == null) {
        resp.setMessage("Search API has no record for Movie with Id " + movieId);
        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(resp);
    }
    record = new Inventory(movieId, DEFAULT_NUM_OF_MOVIES, DEFAULT_NUM_OF_MOVIES);
}
```

File: src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java [hunk 2]
```diff
@@ -242,54 +242,47 @@ public OperationalResponse favoriteMovie(OperationalRequest request) {
...
-					// TODO: Add logs here saying for some reason server was unable to reduce available copies
-					//return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Unable to complete transaction. Please try again later.");
-					return resp;
-				}
+			resp.setMessage("All copies of movie with Id " + movieId + " have been rented to others");
+			return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(resp);
+		}
+		record.setAvailableCopies(record.getAvailableCopies() - 1);
+		if(userRepository.existsById(pk)) {
+			user = userRepository.getOne(pk);
```
Comment (kedarkhetia) [inline src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java]: Same as above over here. 

```
user = userRepository.findById();
if(user != null) {
    ...
}
```

**Review by kedarkhetia — CHANGES_REQUESTED** [review]
Minor changes to save DB calls.

Other than that LGTM.

File: src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java (hunk 1 above)
Comment (mosopeogundipe) [reply src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java]: This is true. My bad. Thanks for pointing it out.

File: src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java (hunk 2 above)
Comment (mosopeogundipe) [reply src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java]: Fixed also

**Review by kedarkhetia — APPROVED** [review]
LGTM.

**Pull request merged**